slasher/db/indexed_attestations.go

```diff
@@ -67,6 +67,35 @@ func (db *Store) IndexedAttestation(targetEpoch uint64, validatorID uint64) ([]*
 	return iAtt, err
 }
 
+// DoubleVotes looks up in db for slashable data possibly multiple slashable actions were preformed by the same validator.
+func (db *Store) DoubleVotes(targetEpoch uint64, validatorIdx uint64, dataroot []byte) ([]*ethpb.AttesterSlashing, error) {
+	idxAttestations, err := db.IndexedAttestation(targetEpoch, validatorIdx)
+	if err != nil {
+		return nil, err
+	}
+	var iAtt []*ethpb.IndexedAttestation
+	var origIdxAtt *ethpb.IndexedAttestation
+	for _, idxAtt := range idxAttestations {
+		root, err := ssz.HashTreeRoot(idxAtt.Data)
+		if err != nil {
+			return nil, err
+		}
+		if !bytes.Equal(root[:], dataroot) {
+			iAtt = append(iAtt, idxAtt)
+		} else {
+			origIdxAtt = idxAtt
```

Comment:
so if the code path never comes here 
```
	var origIdxAtt *ethpb.IndexedAttestation
```
will be nil , and you will be appending it below in the slashing object. You need to pass in the offending attestation as an argument instead of just its data root.

Reply:
thanks, changed